xsdata/writer.py

```diff
@@ -28,20 +30,46 @@ def get_format(self, name) -> AbstractGenerator:
             return self.generators[name]
         raise CodeWriterError(f"Unknown code generator `{name}`")
 
-    def write(self, schema: Schema, classes: List[Class], package: str, format: str):
+    def write(self, classes: List[Class], format: str):
         engine = self.get_format(format)
-        for file, package, output in engine.render(schema, classes, package):
+        for file, package, output in engine.render(classes):
             if len(output.strip()) > 0:
                 logger.info("Generating package: %s", package)
 
                 file.parent.mkdir(parents=True, exist_ok=True)
                 file.write_text(output)
 
-    def print(self, schema: Schema, classes: List[Class], package: str, format: str):
+    def print(self, classes: List[Class], format: str):
         engine = self.get_format(format)
-        for _, _, output in engine.render(schema, classes, package):
+        for _, _, output in engine.render(classes):
             print(output, end="")
 
+    def designate(self, classes: List[Class], format: str):
+        for obj in classes:
+            obj.module = self.unique_module_name(obj.module, format)
+            obj.package = self.unique_package_name(obj.package, format)
+
+    def unique_module_name(self, module: str, format: str):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5141157)